demo/app/src/App.js

```diff
@@ -0,0 +1,349 @@
+      .map(ad => [ad.w, ad.h])
+      .reduce((ads, ad) => ads.concat([ad]), []);
+
+  const buildSetSelection = (setSelected, sizes) => {
+    return event => {
+      const newSelectedSizes = [...sizes];
+      const [currentX, currentY] = event.target.value.split("x");
+
+      newSelectedSizes.forEach(ad => {
+        if ((ad.w === parseInt(currentX)) & (ad.h === parseInt(currentY))) {
+          return (ad.checked = !ad.checked);
+        }
+      });
+      setSelected(newSelectedSizes);
+    };
+  };
+
+  const setSelectionDesktop = buildSetSelection(
+    setSelectedDesktopSizes,
+    selectedDesktopSizes
+  );
+  const setSelectionTablet = buildSetSelection(
+    setSelectedTabletSizes,
+    selectedTabletSizes
+  );
+  const setSelectionMobile = buildSetSelection(
+    setSelectedMobileSizes,
+    selectedMobileSizes
+  );
+
+  const getAdConfiguration = () => {
+    const desktopAds = selectedAds(selectedDesktopSizes);
+    const tabletAds = selectedAds(selectedTabletSizes);
+    const mobileAds = selectedAds(selectedMobileSizes);
+
+    return {
+      path: path,
+      sizes: {
+        mobile: mobileAds,
+        tablet: tabletAds,
+        desktop: desktopAds
+      },
+      targeting: JSON.parse(targeting)
+    };
+  };
+
+  const getBucketConfiguration = () => {
+    let plugins = [];
+
+    if (loggingPlugin) plugins.push("Logging");
+    if (autoRenderPlugin) plugins.push("AutoRender");
+    if (autoRefreshPlugin) plugins.push("AutoRefresh");
+    if (debugPlugin) plugins.push("Debug");
+    if (stickyPlugin) plugins.push("Sticky");
+
+    return {
+      plugins: plugins.map(plugin => window.AdJS.Plugins[plugin]),
+      defaults: {
+      breakpoints: {
+          mobile: { from: 0, to: 350 },
+          tablet: { from: 351, to: 780 },
+          desktop: { from: 781, to: Infinity }
+        },
+        refreshRateInSeconds: refreshRate,
+        offset: offset,
+        targeting: JSON.parse(targeting)
+      }
+    };
+  };
+
+  const setSourceCodePreview = returnCode => {
+    let sourceCode = `
+const mainPage = new AdJS.Bucket(AdJS.Networks.${provider}, ${JSON.stringify(
+      getBucketConfiguration(),
+      null,
+      "  "
+    )});
+
+const el = document.getElementById('ad-slot-1');
+
+const ad = mainPage.createAd(el, ${JSON.stringify(
+      getAdConfiguration(),
+      null,
+      "  "
+    )});
+`;
+
+    setCodePreview(sourceCode);
+  };
+
+  useEffect(() => {
+    setSourceCodePreview();
+    // eslint-disable-next-line
+  }, [
+    path,
+    provider,
+    targeting,
+    refreshRate,
+    offset,
+    autoRefreshPlugin,
+    autoRenderPlugin,
+    debugPlugin,
+    loggingPlugin,
+    responsivePlugin,
+    stickyPlugin,
+    selectedDesktopSizes,
+    selectedTabletSizes,
+    selectedMobileSizes
+  ]);
+
+  const previewAd = () => {
+    if (window.previewInstance) {
+      window.previewInstance.destroy();
+    }
+
+    const el = document.getElementById("ad-slot-1");
```

Comment:
You are also doing this on line 147 and then again block scoped in this method. This should probably be a class level variable.

Reply:
One is to actually assign the element, and the other one is used for code preview as a string.